var/spack/repos/builtin/packages/petsc/package.py

```diff
@@ -140,14 +151,10 @@ def mpi_dependent_options(self):
                 raise RuntimeError('\n'.join(errors))
         else:
             compiler_opts = [
-                '--with-mpi=1',
-                '--with-mpi-dir=%s' % self.spec['mpi'].prefix,
+                '--with-cc=%s' % self.spec['mpi'].mpicc,
+                '--with-cxx=%s' % self.spec['mpi'].mpicxx,
+                '--with-fc=%s' % self.spec['mpi'].mpifc
             ]
-        if sys.platform != "darwin":
-            compiler_opts.extend([
-                '--with-cpp=cpp',
```

Comment:
Yes, the "cpp" passed here may not correspond to the compiler that is used; producing inconsistencies. 

I believe that the problem that required the '--with-cpp=cpp', has been fixed.